src/nvim/generators/gen_api_ui_events.lua

```diff
@@ -128,8 +128,16 @@ for i = 1, #events do
     write_signature(call_output, ev, '')
     call_output:write('\n{\n')
     if ev.remote_only then
+      -- Lua callbacks may emit other events or the same event again. Avoid the latter
+      -- by adding a recursion guard to each generated function that may call a Lua callback.
```

Comment:
I proposed this once before to @bfredl but got no response then, feel free to review it here instead. I think it is more sustainable than adding recursion guards manually where needed and still avoids the problem(only present for Lua callbacks so only adding it here).

Reply:
later on, I think we want recursiveness checks to be more semantic, like at least based on the category of events (like any ext_cmdline event handler shouldn't case another ext_cmdline event), but we can iterate on that during the 0.11 cycle. 

This would be okay for now.